src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -54,8 +54,9 @@ pub struct TargetConfig {
 
 // Returns a mapping of the root package plus its immediate dependencies to
 // where the compiled libraries are all located.
-pub fn compile_targets<'a, 'cfg: 'a>(targets: &[(&'a Target, &'a Profile)],
-                                     pkg: &'a Package,
+pub fn compile_targets<'a, 'cfg: 'a>(pkg_targets: &'a Vec<(&Package,
```

Comment:
`&'a Vec<T>` should basically always be avoided in favor of `&'a [T]` explicitly.